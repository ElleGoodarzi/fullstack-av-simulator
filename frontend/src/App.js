import React, { useState } from 'react';
import axios from 'axios';

function App() {
  const [file, setFile] = useState(null);
  const [result, setResult] = useState(null);
  const [loading, setLoading] = useState(false);

  const handleFileChange = (e) => {
    setFile(e.target.files[0]);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (!file) return;
    setLoading(true);
    const formData = new FormData();
    formData.append('file', file);
    try {
      const response = await axios.post('http://127.0.0.1:5000/upload', formData, {
        headers: { 'Content-Type': 'multipart/form-data' }
      });
      setResult(response.data);
    } catch (error) {
      console.error('Error:', error);
      setResult({ error: 'Upload failed' });
    }
    setLoading(false);
  };

  const simulateSteer = (position) => {
    if (position < -0.2) return 'Steer right';
    if (position > 0.2) return 'Steer left';
    return 'Straight';
  };

  return (
    <div style={{ padding: '20px', maxWidth: '600px', margin: 'auto' }}>
      <h1>AV Lane Detection Simulator</h1>
      <form onSubmit={handleSubmit}>
        <input type="file" onChange={handleFileChange} accept="image/*,video/*" />
        <button type="submit" disabled={loading}>Process</button>
      </form>
      {loading && <p>Processing...</p>}
      {result && (
        <div>
          <p>Lane Position: {result.lane_position.toFixed(4)}</p>
          <p>FPS: {result.fps.toFixed(2)}</p>
          <p>Steering: {simulateSteer(result.lane_position)}</p>
          {result.output_path && (
            <div>
              <h3>Output:</h3>
              <img src={`http://127.0.0.1:5000/${result.output_path}`} alt="Processed" style={{ maxWidth: '100%' }} />
            </div>
          )}
          {result.error && <p>Error: {result.error}</p>}
        </div>
      )}
    </div>
  );
}

export default App;