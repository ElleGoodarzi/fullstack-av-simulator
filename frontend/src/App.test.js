import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import axios from 'axios';
import App from './App';

jest.mock('axios');

test('renders upload form', () => {
  render(<App />);
  expect(screen.getByText(/AV Lane Detection Simulator/i)).toBeInTheDocument();
  expect(screen.getByText(/Process/i)).toBeInTheDocument();
});

test('handles file submit', async () => {
  render(<App />);
  const fileInput = screen.getByLabelText(/file/i);  // If no label, use getByRole('textbox', { type: 'file' })
  const button = screen.getByText(/Process/i);

  fireEvent.change(fileInput, { target: { files: [new File([''], 'test.jpg')] } });
  axios.post.mockResolvedValue({ data: { lane_position: 0.1, fps: 30, output_path: 'output.jpg' } });

  fireEvent.click(button);
  expect(axios.post).toHaveBeenCalledTimes(1);
});

test('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
