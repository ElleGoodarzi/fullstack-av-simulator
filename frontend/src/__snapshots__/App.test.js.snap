// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="padding: 20px; max-width: 600px; margin: auto;"
  >
    <h1>
      AV Lane Detection Simulator
    </h1>
    <form>
      <input
        accept="image/*,video/*"
        type="file"
      />
      <button
        type="submit"
      >
        Process
      </button>
    </form>
  </div>
</DocumentFragment>
`;
